src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -37,8 +37,7 @@ import java.util.*;
 
 }
 
-LEADING_ASTERISK : ( (' '|'\t') {_tokenStartCharPositionInLine == 0}? ) (' '|'\t')* '*'
-      | '*' {_tokenStartCharPositionInLine == 0}?
+LEADING_ASTERISK : (' '|'\t')* '*'
```

Comment:
@rnveach Yes and thus I had to choose from one of the two strategies I mentioned before. I was also wondering that where exactly would my token def and the present token def would create the differences but couldn't find anything and concluded that I shall make a PR and find if there are going to be problems in regression testing.